src/main/java/com/gregtechceu/gtceu/api/cover/CoverBehavior.java

```diff
@@ -220,9 +220,8 @@ public BlockState getAppearance(BlockState sourceState, BlockPos sourcePos) {
         return null;
     }
 
-    @Nullable
-    public IDynamicCoverRenderer getDynamicRenderer() {
-        return null;
+    public Supplier<IDynamicCoverRenderer> getDynamicRendererSupplier() {
```

Comment:
I'd keep the name as `getDynamicRenderer`, this is kind of lengthy
```suggestion
    public @NotNull Supplier<@Nullable IDynamicCoverRenderer> getDynamicRenderer() {
```